src/sidebar/store/modules/groups.js

```diff
@@ -118,6 +119,43 @@ function getGroup(state, id) {
   return state.groups.find(g => g.id === id);
 }
 
+/**
+ * Return groups that don't show up in Featured and My Groups.
+ *
+ * @return {Group|undefined}
+ */
+function getCurrentlyViewingGroups(state) {
+  // If the user is logged in return groups they don't have membership
+  // to and that aren't scoped to the uri.
+  if (selectors.isLoggedIn(state)) {
+    return state.groups.filter(g => !g.isMember && !g.isScopedToUri);
+  }
+  // If the user is logged out return the Public group or groups that
+  // aren't scoped to the uri.
+  return state.groups.filter(g => g.isMember || !g.isScopedToUri);
+}
+
+/**
+ * Return groups the logged in user is a member of.
+ *
+ * @return {Group|undefined}
+ */
+function getMyGroups(state) {
+  if (selectors.isLoggedIn(state)) {
+    return state.groups.filter(g => g.isMember);
+  }
+  return [];
```

Comment:
The need for treating logic for logged-in versus logged-out users here (instead of being able to filter a list of group by membership for both types of users) is because of our magical treatment of the Public group—the client code sets `isMember` to true for all users in all states, which creates these odd side effects. Might be worth commenting.